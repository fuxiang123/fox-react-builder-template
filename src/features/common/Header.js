import React from "react";
import ReactLogo from "@/images/logo.svg";

export default function Header() {
  return (
    <header className="common-header">
      <a className="header-bar" href="/">
        <img src={ReactLogo} alt="React Logo" />
        <div>Fox-Builder</div>
      </a>
    </header>
  );
}
