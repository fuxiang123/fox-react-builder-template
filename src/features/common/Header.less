.common-header {
  position: fixed;
  overflow: hidden;
  top: 0;
  left: 0;
  right: 0;
  padding-left: 50px;
  align-items: center;
  background-color: #20232a;

  .header-bar {
    height: 60px;
    display: flex;
    flex-direction: row;
    align-items: center;
  }

  .header-bar div {
    font-size: 20px;
    color: #61dafb;
    font-weight: 700;
    margin-left: 10px;
  }

  .header-bar img {
    height: 30px;
  }
}
