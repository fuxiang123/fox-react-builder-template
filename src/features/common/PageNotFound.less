.common-page-not-found {
  color: red;
}
