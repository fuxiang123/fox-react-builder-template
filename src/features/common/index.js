// eslint-disable-next-line import/prefer-default-export
export { default as PageNotFound } from "./PageNotFound";
export { default as Header } from "./Header";
