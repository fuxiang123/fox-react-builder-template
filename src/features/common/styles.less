@import "./Header.less";
@import "./PageNotFound.less";
