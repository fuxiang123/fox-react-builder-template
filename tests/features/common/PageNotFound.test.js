import React from "react";
import { render } from "@testing-library/react";
import { PageNotFound } from "../../../src/features/common";

it("common/PageNotFound snapshot", () => {
  const component = render(<PageNotFound />);
  // 快照
  expect(component.asFragment()).toMatchSnapshot();
});
