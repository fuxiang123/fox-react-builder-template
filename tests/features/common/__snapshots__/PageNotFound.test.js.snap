// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common/PageNotFound snapshot 1`] = `
<DocumentFragment>
  <div
    class="common-page-not-found"
  >
    Page not found.
  </div>
</DocumentFragment>
`;
